x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchMaintenanceService.java

```diff
@@ -26,13 +28,17 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import static org.elasticsearch.xpack.search.AsyncSearchIndexService.EXPIRATION_TIME_FIELD;
+import static org.elasticsearch.xpack.search.AsyncSearchIndexService.INDEX;
 
 /**
  * A service that runs a periodic cleanup over the async-search index.
  */
 class AsyncSearchMaintenanceService implements Releasable, ClusterStateListener {
     private static final Logger logger = LogManager.getLogger(AsyncSearchMaintenanceService.class);
 
+    public static final Setting<TimeValue> ASYNC_SEARCH_CLEANUP_INTERVAL_SETTING =
+        Setting.timeSetting("async_search.index_cleanup_interval", TimeValue.timeValueHours(1), Setting.Property.NodeScope);
```

Comment:
do we plan on using this new setting only in our tests? Or can it be useful for users too?

Reply:
Maybe for debug but I don't think it's useful for users, only for tests at the moment.

Reply:
There is no way to somehow make this "private" or for use only in our tests? I worry mostly about naming, and the fact that users may end up using it although we would not want them to.

Reply:
The default value is quite large (1h) so maybe there's some value to change it. I am not sure, I can make it private but that would work only in fantasy integration tests (single vm) so this will come back. I was expecting that the lack of documentation would make this setting an expert thing that users would never heard from ?

Reply:
yea I see, can you add a comment that this is intentionally undocumented?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/55435/commits/09afaca88da489560060ad5a90671652e269006f